### ISSUE TITLE
Cannot pick Wraith King

### ISSUE THREAD

**Issue by dobegor** [EarthSalamander42/dota_imba#878]
It's possible to "pick" it, but the game always starts with a random hero instead.

**Comment by ModderGhost** [comment]
Yes you can by disconnecting before it will random you. Whenever hero is banned with IMBA system (red line on hero, you cannot select it, and hero is not faded into red) you can pick it with dota_select_hero console command.
I guess disconnect works because code is checking if player is actually connected

`if PlayerResource:IsValidPlayer(i) and PlayerResource:GetConnectionState(i) == DOTA_CONNECTION_STATE_CONNECTED then`

#FixIt

**Comment by EarthSalamander42** [comment]
It's not as easy as it looks like. If the server select a hero for you when you're offline, once you reconnect you'll be unable to issue any command. I'll take a closer look to find a way using timers maybe.

**Comment by ModderGhost** [comment]
Do you really need timers? Cannot you hook reconnect event and then check if player actually has valid hero?

**Comment by EarthSalamander42** [comment]
yeah but i'll need timers, iirc there's a delay with the reconnect hook and player being valid
